generators/entity-server/templates/src/main/java/package/repository/search/EntitySearchRepository.java.ejs

```diff
@@ -87,39 +103,62 @@ class <%= entityClass %>SearchRepositoryInternalImpl implements <%= entityClass
 interface <%= entityClass %>SearchRepositoryInternal {
   <%_ if (paginationNo) { _%>
     Stream<<%= persistClass %>> search(String query);
+
+    Stream<<%= persistClass %>> search(Query query);
   <%_ } else { _%>
     Page<<%= persistClass %>> search(String query, Pageable pageable);
+
+    Page<<%= persistClass %>> search(Query query);
   <%_ } _%>
+
+    void index(<%= persistClass %> entity);
 }
 
 class <%= entityClass %>SearchRepositoryInternalImpl implements <%= entityClass %>SearchRepositoryInternal {
 
     private final ElasticsearchRestTemplate elasticsearchTemplate;
+    private final <%= entityClass %>Repository repository;
 
-    <%= entityClass %>SearchRepositoryInternalImpl(ElasticsearchRestTemplate elasticsearchTemplate) {
+    <%= entityClass %>SearchRepositoryInternalImpl(ElasticsearchRestTemplate elasticsearchTemplate, <%= entityClass %>Repository repository) {
         this.elasticsearchTemplate = elasticsearchTemplate;
+        this.repository = repository;
     }
 
   <%_ if (paginationNo) { _%>
     @Override
     public Stream<<%= persistClass %>> search(String query) {
         NativeSearchQuery nativeSearchQuery = new NativeSearchQuery(queryStringQuery(query));
+        return search(nativeSearchQuery);
+    }
+
+    @Override
+    public Stream<<%= persistClass %>> search(Query query) {
         return elasticsearchTemplate
-            .search(nativeSearchQuery, <%= persistClass %>.class)
+            .search(query, <%= persistClass %>.class)
             .map(SearchHit::getContent)
             .stream();
     }
   <%_ } else { _%>
     @Override
     public Page<<%= persistClass %>> search(String query, Pageable pageable) {
         NativeSearchQuery nativeSearchQuery = new NativeSearchQuery(queryStringQuery(query));
-        nativeSearchQuery.setPageable(pageable);
-        List<<%= persistClass %>> hits = elasticsearchTemplate.search(nativeSearchQuery, <%= persistClass %>.class).map(SearchHit::getContent)
+        return search(nativeSearchQuery.setPageable(pageable));
+    }
+
+    @Override
+    public Page<<%= persistClass %>> search(Query query) {
+        List<<%= persistClass %>> hits = elasticsearchTemplate.search(query, <%= persistClass %>.class).map(SearchHit::getContent)
             .stream()
             .collect(Collectors.toList());
 
-        return new PageImpl<>(hits, pageable, hits.size());
+        return new PageImpl<>(hits, query.getPageable(), hits.size());
     }
   <%_ } _%>
+
+    @Override
+    public void index(<%= persistClass %> entity) {
+        repository.find<% if (implementsEagerLoadApis ) { %>OneWithEagerRelationships<% } else { %>ById<% } %>(entity.getId()).ifPresent(elasticsearchTemplate::save);
```

Comment:
```suggestion
        repository.find<% if (implementsEagerLoadApis) { %>OneWithEagerRelationships<% } else { %>ById<% } %>(entity.getId()).ifPresent(elasticsearchTemplate::save);
```

Reply:
Seems to be outdated - can not see the difference

Reply:
Already solved, I committed my suggestion ;)